src/jit/target.h

```diff
@@ -44,15 +44,23 @@
 #error Unsupported or unset target architecture
 #endif
 
-/*****************************************************************************/
-
+//------------------------------------------------------------------------
+//
+// Each register list in register.h must declare REG_STK as the last value.
+// In the following enum declarations, the following REG_XXX are created beyond
+// the "real" registers:
+//    REG_STK          - Used to indicate something evaluated onto the stack.
+//    ACTUAL_REG_COUNT - The number of physical registers. (same as REG_STK).
+//    REG_COUNT        - The number of physical register + REG_STK. This is the count of values that may
+//                       be assigned during register allocation.
+//    REG_NA           - Used to indicate that a register is either not yet assigned or not required.
+//
```

Comment:
Ah, I hadn't intended to include them yet. But `REG_UNK` would be an explicit way to represent that a register is required but not yet assigned, and `REG_OPT` and `REG_OPTUSE` would replace `gtLsraInfo.regOptional`. After `Lowering` `REG_NA` would indicate that no register is required. I'm not sure it's necessary from a functionality perspective, but I think it would make the model cleaner.